File: src/main/java/github/macro/build_info/ClassTag.java
package github.macro.build_info;

import org.jetbrains.annotations.Nullable;

import java.util.Arrays;

/**
 * Created by Macro303 on 2019-Dec-27
 */
public enum ClassTag {
	SCION,
	MARAUDER,
	RANGER,
	WITCH,
	DUELIST,
	TEMPLAR,
	SHADOW;

	@Nullable
	public static ClassTag value(String name) {
		return Arrays.stream(values()).filter(tag -> tag.name().replaceAll("_", " ").equalsIgnoreCase(name)).findFirst().orElse(null);
	}
}

File: src/main/kotlin/github/macro/build_info/ClassTag.kt
package github.macro.build_info

/**
 * Created by Macro303 on 2020-Jan-13.
 */
enum class ClassTag {
    SCION,
    MARAUDER,
    RANGER,
    WITCH,
    DUELIST,
    TEMPLAR,
    SHADOW;

    companion object {
        fun value(name: String): ClassTag? = values().firstOrNull { it.name.equals(name, ignoreCase = true) }
    }
}